src/dispatcher/mod.rs

```diff
@@ -0,0 +1,279 @@
+use crate::clear_console;
+use crate::ItemId;
+use crate::RoomId;
+use std::sync::atomic::{AtomicU8, Ordering};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum ActionId {
+    GoToControlRoom,
+    GoToHall,
+    GoToToneRoom,
+    GoToVocalBooth,
+
+    SpillBeerOnTheConsole,
+
+    PickConsole,
+    PushRandomButton,
+    MakeAllFadersUp,
+    StaringAtTheDeskWithTheDumbFace,
+
+    PickImac,
+    TurnOffImac,
+    BreakScreenOfImac,
+    CloseOpenedProToolsSession,
+    TouchImacDisplayWithFinger,
+
+    PickChair,
+    KickChairFromSoundGuy,
+    PutLegsOnTheChair,
+
+    PickBottle,
+    MakeHoleInBottleAndGetHigh,
+    ThrowBottleToTheAssistant,
+    DrinkFromBottle,
+
+    PickDrums,
+    HitCymbalsRandomly,
+    ScreamAtTheSnare,
+
+    PickGuitarAmp,
+    TurnVolumeToEleven,
+    AllButtonsUp,
+    PressStandBy,
+
+    PickMic,
+    SwearAtTheCleaner,
+    CaughtForThreeMinutesStraight,
+    ThrowMicAtTheWall,
+
+    SingToTheCureAndCry,
+    PressRandomNumbersOnAnIntercom,
+    StaringAtTheDoorForLoongTime,
+
+    DropItem,
+
+    ExitGame,
+}
+
+static MICS_BROKEN: AtomicU8 = AtomicU8::new(0);
+fn break_mic() {
+    MICS_BROKEN.fetch_add(1, Ordering::SeqCst).wrapping_add(1);
+}
+fn get_broken_mics() -> u8 {
+    MICS_BROKEN.load(Ordering::SeqCst)
+}
```

Comment:
Tried to use Cell, faced errors like this might be unsafe. So I decided to end up with any working solution to get thing done, and fix it after feedback

Reply:
Sounds good! I do that as well, first start with an ugly solution and then make it look good